RSDK-11706 —        Update TypeScript SDK with new GetImages signature

hexbabe opened the pull request:
[RSDK-11706](https://viam.atlassian.net/browse/RSDK-11706)


Update TypeScript SDK with new GetImages signature

[RSDK-11706]: https://viam.atlassian.net/browse/RSDK-11706?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hexbabe commented:
@stuqdog @micheal-parks @lia-viam Any ideas on how I can easily manually test this?

micheal-parks commented on src/components/camera/client.ts:
```diff
@@ -59,6 +82,32 @@ export class CameraClient implements Camera {
+      capturedAt: resp.responseMetadata?.capturedAt ?? new Timestamp(),
+    };
+
+    return [images, metadata] as [NamedImage[], ResponseMetadata];
```
```suggestion
    return { images, metadata } 
```
this much more standard than returning an array tuple

hexbabe commented:
> @stuqdog @micheal-parks @lia-viam Any ideas on how I can easily manually test this change? e.g. perhaps spin up a TS client pinned to my local feature branch ver of the SDK. I tried using the node example in this repo and failed miserably on the npm install step and am looking for easier alternatives.

ok I added a package.json bump for the dep that was making my `npm i` fail. I added

```

  const cameraName = 'realsense-1'; // Replace with your camera's name
  const camera = new VIAM.CameraClient(client, cameraName);

  // Now you can use the camera client
  const images = await camera.getImages(['color']);
  console.log('Got images!');
  console.log(images);
```

to main.ts

and saw

```
Got images!
{
  images: [
    {
      sourceName: 'color',
      image: [Uint8Array],
      mimeType: 'image/jpeg'
    }
  ],
  metadata: { capturedAt: S { seconds: 1756394253n, nanos: 328000000 } }
}
```

And when I got rid of the filter, it returned all images

```
Got images!
{
  images: [
    {
      sourceName: 'color',
      image: [Uint8Array],
      mimeType: 'image/jpeg'
    },
    {
      sourceName: 'depth',
      image: [Uint8Array],
      mimeType: 'image/vnd.viam.dep'
    }
  ],
  metadata: { capturedAt: S { seconds: 1756394364n, nanos: 375000000 } }
}
```

micheal-parks commented on src/components/camera/camera.ts:
```diff
@@ -64,6 +76,27 @@ export interface Camera extends Resource {
+   * @param extra - Extra parameters to pass to the camera.
+   */
+  getImages: (
+    filterSourceNames?: string[],
```
Is filterSourceNames bound to a set of constant strings, such as `color`, `depth`, etc?

If so it would be helpful to make that a part of the type:

```suggestion
    filterSourceNames?: ('depth' | 'color')[],
```

hexbabe replied:
no it's not they can be arbitrary

hexbabe commented on src/components/camera/camera.ts:
```diff
@@ -17,9 +18,20 @@ export interface Properties {
+export interface NamedImage {
+  sourceName: string;
+  image: Uint8Array;
+  mimeType: string;
```
Hmm, one of the core new patterns [the scope establishes](https://docs.google.com/document/d/1Q4UWacS76tx3ohOjS98aei2UdHU4XdyWsFldQHTQ_7s/edit?disco=AAABoCWnZb8) is to allow any mime type across the wire. So I don't think we can guarantee a narrow set of mimeType strings. What do you think...

micheal-parks replied:
fine with keeping it a string then 👍
